### PR TITLE
pull consumer add offset save

### PR DESCRIPTION
**Pull request by twz915**
The code implementation mimics the c++ version, depth-first traversing each message queue.
https://github.com/apache/rocketmq-client-cpp/blob/master/example/PullConsumer.cpp

if someone wants to save offset into such as redis, the two method `get_message_queue_offset` and `set_message_queue_offset` could be overrided.

### CODE REVIEW

File: rocketmq/client.py [hunk 1]
```diff
@@ -443,25 +455,33 @@ def pull(self, topic, expression='*', max_num=32):
...
+                elif pull_res.pullStatus == _CPullStatus.NO_NEW_MSG:
+                    has_new_msg = False
+                elif pull_res.pullStatus == _CPullStatus.OFFSET_ILLEGAL:
+                    pass
                 else:
-                    dll.ReleasePullResult(pull_res)  # NOTE: No need to check ffi return code here
-                    break
-                dll.ReleasePullResult(pull_res)  # NOTE: No need to check ffi return code here
+                    pass
+            dll.ReleasePullResult(pull_res)  # NOTE: No need to check ffi return code here
```
Comment (messense) [inline rocketmq/client.py]: Indentation is wrong?

Comment (twz915) [reply rocketmq/client.py]: No, I tested it and it works fine.

Comment (messense) [reply rocketmq/client.py]: It will only release last `pull_res` which leads to memory leak?

Comment (twz915) [reply rocketmq/client.py]: OK, I'll fix this

**Pull request merged**